nomad/rpc_rate_metrics.go

```diff
@@ -0,0 +1,25 @@
+package nomad
+
+import (
+	"github.com/armon/go-metrics"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// MeasureRPCRate increments the appropriate rate metric for this endpoint,
+// with a label from the identity
+func (srv *Server) MeasureRPCRate(
+	endpoint, op string,
+	identity *structs.AuthenticatedIdentity, rpcCtx *RPCContext,
+) {
+	if rpcCtx == nil {
+		return // we're the RPC caller and not the server
+	}
```

Comment:
In some local testing I'm seeing this comment isn't quite accurate; I need to debug this a bit more to verify.

Reply:
So the intention from the RFC was to count whenever an RPC is served by a server or forwarded by a server (because that let's you detect problems like uneven distribution of RPCs from client nodes). If we only cared about metrics on the server that actually serves the request, we could dramatically simplify the problem because auth could happen after checking for forwarding. But I'm pretty sure we do want metrics pre-forwarding?

My intent with this conditional around `rpcCtx == nil` was that we probably don't want to track "static endpoint" RPCs where ex. the deploymentwatcher running on the leader is using RPCs as normal function calls. Unfortunately that also means that if you send a HTTP request to a server, the server that serves the HTTP request won't count it because there's no `rpcCtx` in that case (ex. if the HTTP request hits the leader).

Reply:
> My intent with this conditional around rpcCtx == nil was that we probably don't want to track "static endpoint" RPCs where ex. the deploymentwatcher running on the leader is using RPCs as normal function calls.

Ah that makes sense. I'd love to stop using those static endpoints anyway. I can't imagine there's not a better way to share that logic internally. Perhaps we should file an issue around this (eg `Remove static endpoints to avoid counting them toward metrics` or something) and ship this as-is.